hnn_core/tests/test_params.py

```diff
@@ -79,6 +78,36 @@ def test_base_params():
     assert params == params_base
 
 
+def test_remove_nulled_drives(tmp_path):
+    param_url = ("https://raw.githubusercontent.com/jonescompneurolab/hnn/"
+                 "master/param/ERPYes100Trials.param")
+    params_fname = Path(hnn_core_root, 'param', 'ERPYes100Trials.param')
+    if not op.exists(params_fname):
+        urlretrieve(param_url, params_fname)
+
+    net = jones_2009_model(params=read_params(params_fname),
+                           add_drives_from_params=True,
+                           legacy_mode=True,
+                           )
+    net_removed = remove_nulled_drives(net)
+
+    assert net_removed != net
+
+    # External drives were removed
+    drives_removed = ['bursty1', 'bursty2', 'extgauss', 'extpois']
+    assert all([drive not in net_removed.external_drives.keys()
+                for drive in drives_removed])
+
+    # Connections were removed
+    conn_src_types = set([conn['src_type']
+                          for conn in net_removed.connectivity])
+    assert all([drive not in conn_src_types for drive in drives_removed])
+
+    # gid ranges were updated
```

Comment:
You can also add a check for the `pos_dict`

Reply:
Great, just added that assert.